### PR TITLE
Support for new TS130F shutter switch

### PR DESCRIPTION
**Pull request by Tropaion**
Add support for device discussed in following issue: https://github.com/zigpy/zha-device-handlers/issues/1482

### CODE REVIEW

**Comment by Tropaion** [comment]
dawn it hate these syntax checks...

**Comment by MattWestb** [comment]
Use back playground its fixing the formatting and little more https://black.vercel.app/.

File: zhaquirks/tuya/ts130f.py [hunk 1]
```diff
@@ -332,3 +335,64 @@ class TuyaTS130FTI2(CustomDevice):
             },
         },
     }
+
+class TuyaTS130ESTC(CustomDevice):
```
Comment (javicalle) [inline zhaquirks/tuya/ts130f.py:339]: 2 breaklines before the class definition

**Comment by Tropaion** [comment]
Is there a web page where ich can run/check isort? I only found this: https://pycqa.github.io/isort/docs/quick_start/0.-try.html but I didn't find which config is used here.

**Comment by javicalle** [comment]
Not tested:
* https://pycqa.github.io/isort/docs/quick_start/0.-try.html
* 
If your are running on Linux you can install `black` and `isort` (better  in an virtual environment) and run from command line. On the long run you can get used to it...

**Comment by MattWestb** [comment]
Move the line `from zhaquirks.tuya import TuyaZBExternalSwitchTypeCluster`
to after this
```
from zhaquirks.const import (
    DEVICE_TYPE,
    ENDPOINTS,
    INPUT_CLUSTERS,
    MODEL,
    OUTPUT_CLUSTERS,
    PROFILE_ID,
)
```
And before:
```
ATTR_CURRENT_POSITION_LIFT_PERCENTAGE = 0x0008
CMD_GO_TO_LIFT_PERCENTAGE = 0x0005
```
with one blank line before the attribute and after the last import..

**Comment by javicalle** [comment]
```python
"""Device handler for loratap TS130F smart curtain switch."""
from zigpy.profiles import zha
from zigpy.quirks import CustomCluster, CustomDevice
import zigpy.types as t
from zigpy.zcl.clusters.closures import WindowCovering
from zigpy.zcl.clusters.general import (
    Basic,
    GreenPowerProxy,
    Groups,
    OnOff,
    Ota,
    Scenes,
    Time,
)

from zhaquirks.const import (
    DEVICE_TYPE,
    ENDPOINTS,
    INPUT_CLUSTERS,
    MODEL,
    OUTPUT_CLUSTERS,
    PROFILE_ID,
)
from zhaquirks.tuya import TuyaZBExternalSwitchTypeCluster

ATTR_CURRENT_POSITION_LIFT_PERCENTAGE = 0x0008
CMD_GO_TO_LIFT_PERCENTAGE = 0x0005
```

**Comment by MattWestb** [comment]
One empty line need being  deleted:
```
     OUTPUT_CLUSTERS,
     PROFILE_ID,
 )
-
 from zhaquirks.tuya import TuyaZBExternalSwitchTypeCluster
```
(The  ` -  ` line)

**Comment by Tropaion** [comment]
Managed to install isort in Visual Studio Code, maybe it works better now :D

**Comment by MattWestb** [comment]
BINGO !!!!

**Comment by TonkaNL** [comment]
Hi all,

I'm having an issue with my set of TS130F's.

https://community.home-assistant.io/t/ts130f-zigbee-window-cover-module-unresponsive-because-of-green-power-proxy/412867

I'm new to github and also not very proficient in modifying quirks. I really want to learn what's going on. Can anyone point me in the right direction for me to read up on this topic? Any help would be much appreciated.

Best regards,

Tonka

**Comment by Tropaion** [comment]
@TonkaNL Did you try this quirk I created here? With this one, your device should work, since it also uses GP, looks like it's the same: https://github.com/Tropaion/zha-device-handlers/blob/1073a0a66531d05f39f9042c85512ea1f52e5941/zhaquirks/tuya/ts130f.py

**Comment by TonkaNL** [comment]
Hi Tropaion,

Thanks. So how do I do this?

And how do I specifically assign your quirk to the one divice that does not work. I don't want to break the working one..

**Comment by Tropaion** [comment]
@TonkaNL You won't break the existing one and when you add it correctly it applies automatically.
Like this: https://github.com/zigpy/zha-device-handlers/discussions/693#discussioncomment-857274

**Comment by TonkaNL** [comment]
Easy peasy. Quirk is in use. Thanks!

However..  the device still can not be controlled by HA. When the buttons are used, it works and shows it's activity and position in realtime in HA. But when I want to control the cover with HA, nothing happens. Hmm.

**Comment by TonkaNL** [comment]
I have removed and re-paired the device. It always works for a while. Let's see what happens.

**Comment by TonkaNL** [comment]
Well...  seems to be working allright!  I'll let you know if it stays this way. Thanks a bunch!

**Comment by TonkaNL** [comment]
Still works like a charm. With the previous quirk it would have gone haywire already.

**Comment by Tropaion** [comment]
@TonkaNL Ok, thats good. Did you add the device manufacturer code to the quirk or is it the same as in the quirk?
Because if you changed/added the quirk we have to make a PR.

**Comment by TonkaNL** [comment]
Nope. Just dropped the quirk in the folder and it worked. After a reboot of Hassio core, the new quirks were visible in the config. The device that worked, worked. The one that did not, I had to repair.. re-pair 😉.

Now everything is fully functional again.

**Pull request merged**